Felienne/hedy
# Setup black formatter on pre-commit for python formatting

**Description**
Added pre-commit configuration to run [Black](https://github.com/psf/black) when committing code. This should make code-reviews much easier and keep styling consistent across the repository.

**Fixes issue or discussion number**
No linked issue, discussed in thread on Discord

**How to test**
Since black has run on many many files, the changes are too big to code-review. I suggest you reproduce the following steps on the main branch instead, and then merge my pull request into it to verify I haven't changed anything else than the formatting.

- Install pre-commit using pip (`pip install pre-commit`)
- Add pre-commit to `requirements.txt`
- Add `pre-commit-config.yaml` with the same values as in my branch
- Run pre-commit install to install the Black package for pre-commit
- Run `pre-commit run --all-files` to run Black across all python files in the directory, instead of just when changes are made to a file. This is a one-time operation

**Note**
Would be good to include formatting for Jinja templates, Typescript, etc. Didn't want to make the choice on _what_ formatter to use. You can switch out Black for Autopep8 if that's desired. Another good addition would be running Flake8 on pre-commit for python linting. Similar linters can be used for other languages (Typescript, HTML, CSS, etc.) to improve the code consistency and quality.
